src/user/user.controller.ts

```diff
@@ -1,4 +1,87 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Get, HttpStatus, Param, ParseIntPipe, Post, Res } from '@nestjs/common';
+import { TransactionType } from '@prisma/client';
+import { TransactionService } from 'src/transactions/transactions.service';
+import { UserService } from './user.service';
+import { ProviderService } from 'src/provider/provider.service';
+import { PurchaseDto } from './dto/purchase.dto';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { WalletCredits } from 'src/wallet/dto/wallet.dto';
+import { Transaction } from 'src/transactions/dto/transactions.dto';
 
-@Controller('user')
-export class UserController {}
+@ApiTags('users')
+@Controller('users')
+export class UserController {
+    constructor(
+        private transactionService: TransactionService,
+        private userService: UserService,
+        private providerService: ProviderService
+    ) {}
+
+    @ApiOperation({ summary: 'Get User Credits' })
+    @ApiResponse({ status: HttpStatus.OK, description: 'Credits fetched successfully', type: WalletCredits })
+    @Get("/:userId/credits")
+    // get credits of a particular user
+    async getCredits(
+        @Param("userId", ParseIntPipe) userId: number
+    ) {
+        // fetch wallet
+        const wallet = await this.userService.getUserWallet(userId);
+
+        return {
+            statusCode: HttpStatus.OK,
+            message: "Credits fetched successfully",
+            body: {
+                credits: wallet.credits
+            }
+        }
+    }
+
+    @ApiOperation({ summary: 'Get User Transactions' })
+    @ApiResponse({ status: HttpStatus.OK, description: 'transactions fetched successfully', type: [Transaction] })
+    @Get("/:userId/transactions")
+    // get all transactions of a particular user
+    async getUserTransactions(
+        @Param("userId", ParseIntPipe) userId: number
+    ) {
+        // check user
+        await this.userService.getUserWallet(userId);
+
+        // fetch transactions
+        const transactions = await this.transactionService.fetchTransactionsOfOneSystemActor(userId);
+        return {
```

Comment:
Use built-in @Res() decorator to send responses instead of manually constructing response objects